From 5f3ceedc9c08a77aacb3c3e0b1d09afb61329be0 Mon Sep 17 00:00:00 2001
From: Mitch Curtis <mitch.curtis@qt.io>
Date: Wed, 29 Jul 2020 13:41:11 +0200
Subject: [PATCH] Remove operator== for QInputMethodEvent::Attribute

This was added to qtbase in 46ec92dad0f71c8d5e3260a67321d82054ea607e.

Task-number: QTBUG-85789
Change-Id: Ide246ba7bc75d4ef51d2c1f35172b64bece332c1
---
 src/virtualkeyboard/qvirtualkeyboardinputcontext_p.cpp | 8 --------
 src/virtualkeyboard/shadowinputcontext.cpp             | 1 -
 2 files changed, 9 deletions(-)

diff --git a/src/virtualkeyboard/qvirtualkeyboardinputcontext_p.cpp b/src/virtualkeyboard/qvirtualkeyboardinputcontext_p.cpp
index 4365178..94c6283 100644
--- a/src/virtualkeyboard/qvirtualkeyboardinputcontext_p.cpp
+++ b/src/virtualkeyboard/qvirtualkeyboardinputcontext_p.cpp
@@ -44,14 +44,6 @@
 
 QT_BEGIN_NAMESPACE
 
-bool operator==(const QInputMethodEvent::Attribute &attribute1, const QInputMethodEvent::Attribute &attribute2)
-{
-    return attribute1.start == attribute2.start &&
-           attribute1.length == attribute2.length &&
-           attribute1.type == attribute2.type &&
-           attribute1.value == attribute2.value;
-}
-
 using namespace QtVirtualKeyboard;
 
 const bool QtVirtualKeyboard::QT_VIRTUALKEYBOARD_FORCE_EVENTS_WITHOUT_FOCUS = qEnvironmentVariableIsSet("QT_VIRTUALKEYBOARD_FORCE_EVENTS_WITHOUT_FOCUS");
diff --git a/src/virtualkeyboard/shadowinputcontext.cpp b/src/virtualkeyboard/shadowinputcontext.cpp
index dc49f51..75fa479 100644
--- a/src/virtualkeyboard/shadowinputcontext.cpp
+++ b/src/virtualkeyboard/shadowinputcontext.cpp
@@ -36,7 +36,6 @@
 #include <QQuickItem>
 
 QT_BEGIN_NAMESPACE
-bool operator==(const QInputMethodEvent::Attribute &attribute1, const QInputMethodEvent::Attribute &attribute2);
 
 namespace QtVirtualKeyboard {
 
